docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,484 @@
+provides a launcher command that works with the trainer and handles all
+the ``torch.distributed`` setup for you.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                                       device='gpu')
+   trainer.fit()
+
+Access the Composer launcher via the ``composer`` command along with the
+number of GPUs you'd like to use and your training script. Use
+``composer --help`` to see a full list of configurable options.
+
+.. code:: bash
+
+   # run training on 8 GPUs
+   >>> composer -n 8 run_trainer.py
+
+.. seealso::
+
+    Our :doc:`Distributed Training<trainer/distributed_training>` guide and
+    the :mod:`composer.utils.dist` module.
+
+
+DeepSpeed Integration
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer comes with DeepSpeed support, allowing you to leverage their
+full set of features that makes it easier to train large models across
+(1) any type of GPU and (2) multiple nodes. For more details on DeepSpeed,
+see `their website <https://www.deepspeed.ai>`__.
+
+To enable DeepSpeed, simply pass in a config as specified in the
+DeepSpeed docs `here <https://www.deepspeed.ai/docs/config-json/>`__.
+
+.. code:: python
+
+   # run_trainer.py
+
+   from composer import Trainer
+
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu',
+                     deepspeed_config={
+                         "train_batch_size": 2048,
+                         "amp": {"enabled": True},
+                     })
+
+
+.. warning::
+
+    The ``deepspeed_config`` must not conflict with any other parameters
+    passed to the trainer.
+
+
+Callbacks
+~~~~~~~~~
+
+You can insert arbitrary callbacks to be run at various points during
+the training loop. The Composer library provides several useful
+callbacks for things such as monitoring throughput and memory usage
+during training, but you can also implement your own.
+
+.. code:: python
+
+   from composer import Trainer
+   from composer.callbacks import SpeedMonitor
+
+   # include a callback for tracking throughput/step during training
+   trainer = Trainer(model=model,
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                     max_duration='160ep',
+                     device='gpu',
+                     callbacks=[SpeedMonitor(window_size=100)])
+
+.. seealso::
+
+    The :doc:`Callbacks<trainer/callbacks>` guide and :mod:`composer.callbacks`.
+
+
+Numerics
+~~~~~~~~
+
+Using `mixed precision <https://arxiv.org/abs/1710.03740>`__ can speed up your training loop and only requires
```

Comment:
Can we add a blurb about what mixed precision is, and/or link to the pytorch docs for it? The paper is a bit intense...